### PR TITLE
Migrate from flake8/isort to ruff

### PR DESCRIPTION
**Pull request by ChrisLovering**

### CODE REVIEW

**Review by janine9vn — APPROVED** [review]
woof woof

File: bot/services.py [hunk 1]
```diff
@@ -43,17 +43,18 @@ async def send_to_paste_service(contents: str, *, extension: str = "") -> str |
...
 
-            if extension == '.py':
-                return paste_link
+        paste_link = f"{PASTE_URL}/{response_json['key']}{extension}"
 
-            return paste_link + "?noredirect"
+        if extension == ".py":
+            return paste_link
+
+        return paste_link + "?noredirect"
```
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply bot/services.py:54]: The `if` block above returns, the the `elif` isn't needed

File: bot/utils/checks.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ async def predicate(ctx: commands.Context) -> bool | NoReturn:
         if ctx.message.channel.category.name == code_jam_category_name:
             return True
         log.trace(f"{ctx.author} tried to invoke {ctx.command.name} outside of the Code Jam categories.")
-        raise CodeJamCategoryCheckFailure()
+        raise CodeJamCategoryCheckFailure
```
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply bot/utils/checks.py:24]: It's been like for all of Python 3.X. 2.7 behaved this way too.

https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement
> If it is a class, the exception instance will be obtained when needed by instantiating the class with no arguments.

File: tests/test_code_jam.py [hunk 3]
```diff
@@ -129,7 +137,7 @@ async def test_channel_overwrites(self):
     @autospec(_creation_utils, "_add_team_leader_roles", pass_mocks=False)
     async def test_team_channels_creation(self, get_category, get_overwrites):
         """Should create a text channel for a team."""
-        team_leaders = MockRole()
+        MockRole()
```
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply tests/test_code_jam.py]: Tests pass with it removed, so I removed it, nice spot. https://github.com/python-discord/sir-robin/compare/47a29a597336cb1766943fae83360f0afa1f10ab..e1346ad83b7f4c5b9dc16aca559a3d6d6472b801

File: bot/services.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply bot/services.py]: I've just realised this file is for the old paste service anyway, which I'm going to delete in my next PR and it's entirely non-functional right now with the old paste service being offline

**Pull request merged**